pkg/controller/tailoredprofile/tailoredprofile_controller.go

```diff
@@ -187,6 +261,280 @@ func (r *ReconcileTailoredProfile) Reconcile(ctx context.Context, request reconc
+			}
+			newRules = append(newRules, *rule)
+		}
+		if len(newRules) != len(v1alphaTp.Spec.DisableRules) {
+			v1alphaTpCP.Spec.DisableRules = newRules
+		}
+	}
+
+	// check if there are any enabled rules that are migrated
+	if len(v1alphaTp.Spec.EnableRules) > 0 {
+		var newRules []cmpv1alpha1.RuleReferenceSpec
+		for ri := range v1alphaTp.Spec.EnableRules {
+			rule := &v1alphaTp.Spec.EnableRules[ri]
+			if checkType, ok := migratedRules[rule.Name]; ok && checkType != string(profileType) {
+				doContinue = false
+				logger.Info("Removing migrated rule from enableRules", "rule", rule.Name)
+				r.Eventf(v1alphaTp, corev1.EventTypeWarning, "TailoredProfileMigratedRule", "Removing migrated rule %s from enableRules", rule.Name)
+				continue
+			}
+			newRules = append(newRules, *rule)
+		}
+
+		if len(newRules) != len(v1alphaTp.Spec.EnableRules) {
+			v1alphaTpCP.Spec.EnableRules = newRules
+		}
+		if v1alphaTp.Spec.Extends == "" && len(newRules) == 0 && len(v1alphaTpCP.Spec.DisableRules) == 0 && !doContinue {
+			// flip the checkType of the TailoredProfile to the new checkType if the TailoredProfile does not extend any profile
+			// and it does not have any rules left after removing migrated rules and the before migration TailoredProfile has
+			// enableRules
+			if profileType == cmpv1alpha1.ScanTypeNode {
+				v1alphaTpCP.SetAnnotations(map[string]string{cmpv1alpha1.ProductTypeAnnotation: string(cmpv1alpha1.ScanTypePlatform)})
+			} else {
+				v1alphaTpCP.SetAnnotations(map[string]string{cmpv1alpha1.ProductTypeAnnotation: string(cmpv1alpha1.ScanTypeNode)})
+			}
+			v1alphaTpCP.Spec.EnableRules = v1alphaTp.Spec.EnableRules
+			logger.Info("Flipping the checkType of the TailoredProfile to the new checkType", "checkType", profileType)
+			r.Eventf(v1alphaTp, corev1.EventTypeWarning, "TailoredProfileMigratedCheckType", "Flipping the checkType of the TailoredProfile to the new checkType: %s", profileType)
+		}
+	}
+
+	if v1alphaTp.Spec.Extends == "" && len(v1alphaTpCP.Spec.DisableRules) == 0 && len(v1alphaTpCP.Spec.EnableRules) == 0 {
+		errorMsg := "TailoredProfile does not have any rules left after removing migrated rules and it does not extend any profile"
+		v1alphaTpCP.Status.State = cmpv1alpha1.TailoredProfileStateError
+		v1alphaTpCP.Status.ErrorMessage = errorMsg
+		doContinue = false
+		logger.Info(errorMsg)
+		r.Eventf(v1alphaTp, corev1.EventTypeWarning, "TailoredProfileMigratedRule", errorMsg)
+	}
+
+	if !doContinue {
+		err = r.Client.Update(context.TODO(), v1alphaTpCP)
+		logger.Info("Updating TailoredProfile with migrated rules removed")
+		if err != nil {
+			return false, err
+		}
+	}
+	return doContinue, nil
+}
+
+func isValidationRequired(tp *cmpv1alpha1.TailoredProfile) bool {
+	if tp.Spec.Extends != "" {
+		if tp.Spec.DisableRules != nil || tp.Spec.EnableRules != nil || tp.Spec.ManualRules != nil || tp.Spec.SetValues != nil {
+			return true
+		}
+	} else {
+		if tp.Spec.EnableRules != nil || tp.Spec.ManualRules != nil {
```

Comment:
nit: These are duplicated from the conditional check on line 356. We could simplify them by doing this check first.

We can do this in a follow up.

Reply:
maybe doing something like:

```
func isValidationRequired(tp *cmpv1alpha1.TailoredProfile) bool {
	if tp.Spec.Extends != "" {
		return tp.Spec.DisableRules != nil || tp.Spec.EnableRules != nil || tp.Spec.ManualRules != nil || tp.Spec.SetValues != nil
	}
	return tp.Spec.EnableRules != nil || tp.Spec.ManualRules != nil
}
```